server/server.js

```diff
@@ -84,7 +88,195 @@ app.get('*', (req, res) => {
   res.sendFile(path.join(__dirname, '/../react-client/dist', 'index.html'));
 });
 
-//  POST requests
+//  POST request
+app.post('/addReview/:productId', (req, res) => {
+  const { productId } = req.params;
+  // If I don't stringify req.body and then console.log req.body I get [object, Object]
+  //  I don't understand why but if I stringify, then parse it works fine.
+  //  TODO figure out why this is happening and fix it
+  let reqBody = JSON.stringify(req.body);
+  reqBody = JSON.parse(reqBody);
+  const query = productId.toUpperCase();
+
+  //  I was not able to figure out findOneAndUpdate when pushing to an array.
+  //  TODO refactor to use findOneAndUpdate
+  Reviews.findOne(
+    { product_id: query },
+    (err, result) => {
+      if (err) {
+        console.log(`Error querying database for product # ${query}`);
+        console.error(err);
+        res.sendStatus(404);
+      } else {
+        result.reviews.push(reqBody);
+        result.save((saveError, updatedReviews) => {
+          if (saveError) {
+            console.error(saveError);
+          } else {
+            const { reviews } = updatedReviews;
+            //  log last insertion
+            console.log(reviews[reviews.length - 1]);
+            res.sendStatus(200);
+          }
+        });
+      }
+    },
+  );
+});
+
+// Update Reviews routes - 1 per review field
+app.patch('/updateReviewlikes/:productId', (req, res) => {
```

Comment:
To confirm re: review likes - is the client sending an updated # of likes, or would it just increment the existing number of likes when the client clicks the like button?

Reply:
I believe it's both - when the button is clicked the client would update via this route. I haven't really looked at Server Side rendering yet so it's possible that might change things.